x-pack/solutions/observability/plugins/observability_onboarding/ui_tests/tests/custom_logs/configuration.spec.ts

```diff
@@ -0,0 +1,257 @@
+  }) => {
+    await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+
+    await expect(customLogsPage.serviceNameInput()).toHaveValue('');
+    await expect(customLogsPage.continueButton()).not.toBeDisabled();
+  });
+
+  test('Users should expand and collapse the Advanced settings section', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await expect(customLogsPage.namespaceInput()).not.toBeInViewport();
+    await expect(customLogsPage.customConfigInput()).not.toBeInViewport();
+
+    await customLogsPage.clickAdvancedSettingsButton();
+
+    await expect(customLogsPage.namespaceInput()).toBeInViewport();
+    await expect(customLogsPage.customConfigInput()).toBeInViewport();
+
+    await customLogsPage.clickAdvancedSettingsButton();
+
+    await expect(customLogsPage.namespaceInput()).not.toBeInViewport();
+    await expect(customLogsPage.customConfigInput()).not.toBeInViewport();
+  });
+
+  test('Users should see a default namespace', async ({ pageObjects: { customLogsPage } }) => {
+    customLogsPage.clickAdvancedSettingsButton();
+
+    await expect(customLogsPage.namespaceInput()).toHaveValue('default');
+  });
+
+  test('Users should not be able to continue if they do not specify a namespace', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    customLogsPage.clickAdvancedSettingsButton();
+
+    await customLogsPage.namespaceInput().fill('');
+    await expect(customLogsPage.continueButton()).toBeDisabled();
+  });
+
+  test('should be optional allowing user to continue if it is empty', async ({
+    pageObjects: { customLogsPage },
+  }) => {
+    await customLogsPage.clickAdvancedSettingsButton();
+    await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+    await expect(customLogsPage.customConfigInput()).toHaveValue('');
+    await expect(customLogsPage.continueButton()).not.toBeDisabled();
+  });
+
+  test.describe('Integration name', () => {
+    test.beforeEach(async ({ pageObjects: { customLogsPage } }) => {
+      await customLogsPage.logFilePathInput(0).fill('myLogs.log');
+    });
+
+    test('Users should not be able to continue if they do not specify an integrationName', async ({
+      pageObjects: { customLogsPage },
+    }) => {
+      await customLogsPage.integrationNameInput().fill('');
+      await expect(customLogsPage.continueButton()).toBeDisabled();
+    });
+
+    test('value will contain _ instead of special chars', async ({
+      pageObjects: { customLogsPage },
+    }) => {
+      await customLogsPage.integrationNameInput().fill('hello$world');
+      await expect(customLogsPage.integrationNameInput()).toHaveValue('hello_world');
+    });
+
+    test('value will be invalid if it is not lowercase', async ({
+      pageObjects: { customLogsPage },
+      page,
+    }) => {
+      await customLogsPage.integrationNameInput().fill('H3llowOrld');
+      await expect(page.getByText('An integration name should be lowercase.')).toBeVisible();
```

Comment:
nit: messages could be extracted out as in https://github.com/elastic/kibana/blob/c2d20d3b4e4cfafb5216fc36484b2446ba591255/x-pack/platform/plugins/private/discover_enhanced/ui_tests/tests/value_suggestions.spec.ts#L45

Reply:
I've put those messages into a static property of the page object, seems like a mo contextual place for them